account_invoice_transmit_method/models/transmit_method.py

```diff
@@ -18,9 +19,48 @@ class TransmitMethod(models.Model):
         string='Selectable on Customers', default=True)
     supplier_ok = fields.Boolean(
         string='Selectable on Vendors', default=True)
+    send_mail = fields.Boolean()
+    mail_template_id = fields.Many2one(
+        'mail.template',
+        domain="[('model', '=', 'account.invoice')]"
+    )
 
     _sql_constraints = [(
         'code_unique',
         'unique(code)',
         'This transmit method code already exists!'
     )]
+
+    def _update_send_mail_composer(self, invoice, template, composer_form):
+        # We create this as a hook if someone wants to add encryption,
+        # extra attachments or other stuff
+        if template:
+            composer_form.template_id = template
+
+    def _transmit_send_invoice(self, invoice):
+        template = self.mail_template_id or self.env.ref(
+            'account.email_template_edi_invoice', False)
+        lang = self.env.context.get('lang')
+        if template and template.lang:
+            lang = template._render_template(
+                template.lang, 'account.invoice', self.id)
+        self = self.with_context(lang=lang)
+        with Form(self.env['mail.compose.message'].with_context(
```

Comment:
My idea here is to use the code upstream for doing this (and avoid duplicating code - DRY principle), so can you change this to create a new send transient record with corresponding active_ids that send the mail?

Reply:
@pedrobaeza I don't understand what do you want exactly :thinking: 
Do you want to open the mail.compose.message in order to change it?
I though that the main idea was to send the mail directly with intervention of the user. Also, If there are more than one transmit method, this could be a problem

Reply:
I don't want to do it interactively, just reuse the existing transient model that do the dirty stuff for you (like render_template).

Reply:
ok!!! I understand now, thanks :wink:

Reply:
I think this is what you expected :wink: